components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,116 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:PrizePool
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Opponent = require('Module:Opponent')
+local Table = require('Module:Table')
+
+local PrizePool = Class.new(function(self, ...) self:init(...) end)
+
+-- TODO: Extract Enum handling to its own Module
+local prizeTypes = Table.map({
+    'USD', 'LOCAL_CURRENCY', 'POINTS', 'QUALIFIES', 'FREETEXT'
+}, function(value, name) return name, value end)
+
+-- TODO: Add possiblity to add custom prizeData
+local prizeData = {
+    [prizeTypes.USD] = {row = 'usdprize'},
+    [prizeTypes.LOCAL_CURRENCY] = {header = 'localcurrency', row = 'localprize'},
+    [prizeTypes.POINTS] = {header = 'points', row = 'points'},
+    [prizeTypes.QUALIFIES] = {header = 'qualifies', row = 'qualified'},
+    [prizeTypes.FREETEXT] = {header = 'freetext', row = 'freetext'},
+}
+
+function PrizePool:init(args)
+    self.args = args
+    self.pagename = mw.title.getCurrentTitle().text
+    self.args.type = Json.parseIfString(self.args.type)
+    self.widgetInjector = nil
+    self.lpdbInjector = nil
+
+    if not self.args.type then
```

Comment:
the type handling on sc/sc2 would need some specifics to be able to convert e.g. archon to duo plus some (later to be stored in extradata) param that indicates it is archon

Reply:
Should be possible to handle this from the customs.